pkg/kubefedctl/federate/util.go

```diff
@@ -168,6 +172,10 @@ func apiResourceMatchesSkipName(apiResource metav1.APIResource, skipAPIResourceN
 	return false
 }
 
+func apiResourceNameMatchesInvalidCRDResourceNames(apiResource metav1.APIResource) bool {
+	return strings.ContainsAny(apiResource.Name, forbiddenCharsInDNS1123Subdomain)
```

Comment:
Instead of tracking which characters are forbidden in the DNS 1123 spec, I suggest we just call:
https://github.com/kubernetes-sigs/kubefed/blob/fc7a5fd38f4192f9ef802017b3fe58ef9045e5de/vendor/k8s.io/apimachinery/pkg/util/validation/validation.go#L133-L144
and return true if errors are reported.

Reply:
@font That's an excellent suggestion. I did a cursory grep for such a function and had missed this somehow. Maybe didn't have the right grep search string. In either case, I have made the changes that use the above.